driver/src/orderbook/streamed/state.rs

```diff
@@ -0,0 +1,561 @@
+                // the exchange because tokens can be deposited anyway.
+                let token_id = self.tokens.id(*token_address)?;
+                Some((
+                    (*user_id, token_id),
+                    balance.current_balance(batch_id).low_u128(),
+                ))
+            })
+    }
+
+    pub fn orders(&self, batch_id: BatchId) -> impl Iterator<Item = ModelOrder> + '_ {
+        self.orders
+            .iter()
+            .filter(move |(_, order)| order.valid_from <= batch_id && batch_id <= order.valid_until)
+            .map(|((user_id, order_id), order)| {
+                let (buy_amount, sell_amount) = stablex_auction_element::compute_buy_sell_amounts(
+                    order.price_numerator,
+                    order.price_denominator,
+                    order.price_denominator - order.used_amount,
+                );
+                ModelOrder {
+                    id: *order_id,
+                    account_id: *user_id,
+                    buy_token: order.buy_token,
+                    sell_token: order.sell_token,
+                    buy_amount,
+                    sell_amount,
+                }
+            })
+    }
+
+    /// Reset the state to the default state in which no events have been applied.
+    pub fn clear(&mut self) {
+        self.orders.clear();
+        self.balances.clear();
+        self.tokens.0.clear();
+    }
+
+    /// Apply an event to the state, modifying it.
+    ///
+    /// In case of error no modifications take place.
+    ///
+    /// `block_batch_id` is the current batch based on the timestamp of the block that contains the
+    ///  event.
+    pub fn event(&mut self, event: &Event, block_batch_id: BatchId) -> Result<(), Error> {
+        match event {
+            Event::Deposit(event) => self.deposit(event, block_batch_id),
+            Event::WithdrawRequest(event) => self.withdraw_request(event),
+            Event::Withdraw(event) => self.withdraw(event, block_batch_id),
+            Event::TokenListing(event) => self.token_listing(event),
+            Event::OrderPlacement(event) => self.order_placement(event),
+            Event::OrderCancellation(event) => self.order_cancellation(event, block_batch_id),
+            Event::OrderDeletion(event) => self.order_deletion(event),
+            Event::Trade(event) => unimplemented!(),
+            Event::TradeReversion(event) => unimplemented!(),
+            Event::SolutionSubmission(event) => self.solution_submission(event, block_batch_id),
+        }
+    }
+
+    fn deposit(&mut self, event: &Deposit, block_batch_id: BatchId) -> Result<(), Error> {
+        self.balances
+            .entry((event.user, event.token))
+            .or_default()
+            .deposit(
+                Flux {
+                    amount: event.amount,
+                    batch_id: event.batch_id,
+                },
+                block_batch_id,
```

Comment:
The contract currently only emits deposit events with batch_id set to getCurrentBatchId, so yes. On the other hand the event still does have this field which indicates it might be possible to set up a deposit further in the future.
Should I add a check and turn it into an error?